### PR TITLE
Improve flow of readme

### PR DESCRIPTION
**Pull request by richlander**
Changed the ordering of content and a few pieces of wording. Primary goal was to pop contributing opportunities as early as possible in the readme, since the components list is getting quite long. We may want to do something different with the list as it grows even longer, but we'll not worry about that now.

### CODE REVIEW

**Comment by stephentoub** [comment]
LGTM, though you might want to commit this to v1.0 rather than to master.

**Comment by richlander** [comment]
Happy to do that, so that it updates the v1 readme, too. I assume that's the goal.

Any more feedback before I do that, say from @terrajobst?

**Comment by stephentoub** [comment]
> I assume that's the goal.

Yes, but also because I think it's mostly being updated in v1.0, in particular as new libraries are committed to v1.0, so it'll reduce the chances of conflicts.  Not a big deal, though.

File: README.md [hunk 1]
```diff
@@ -9,13 +9,37 @@
...
+
+Some of the best ways to contribute are to try things out, file bugs, and join in design conversations. 
+
+Want to get more familiar with what's going on in the code?
+* [Pull requests](https://github.com/dotnet/corefx/pulls): [Open](https://github.com/dotnet/corefx/pulls?q=is%3Aopen+is%3Apr)/[Closed](https://github.com/dotnet/corefx/pulls?q=is%3Apr+is%3Aclosed)
+
+Looking for something to work on? The list of [up-for-grabs issues](https://github.com/dotnet/corefx/issues?q=is%3Aopen+is%3Aissue+label%3Aup-for-grabs) is a great place to start.
+
+* [How to Contribute][Contributing Guide]
+    * [Contributing Guide][Contributing Guide]
```
Comment (weshaggard) [inline README.md]: Why the double link to Contributing Guide?

Comment (weshaggard) [inline README.md]: Actually there are 4 links to it in this section :)

**Comment by weshaggard** [comment]
LGTM but I agree with @stephentoub that we should update this in the v1.0 branch.

**Comment by richlander** [comment]
@weshaggard - great. I'll do that later today. 

Don't know how I ended up with multiple links. That's very strange. I'll make sure that those are not there when I PR to the other branch.

**Comment by richlander** [comment]
@weshaggard - I took a closer look at the contributing links. Those were already present in master.

https://github.com/dotnet/corefx/blob/master/README.md

**Comment by richlander** [comment]
Closing in favor of PR #596.

**Pull request closed without merging**